load_data.js

```diff
@@ -0,0 +1,73 @@
+/*jshint esversion: 6 */
+(function(window){
+  window.extractData = function() {
+    const ret = $.Deferred();
+    
+    function isLeapYear(year) {
+      return new Date(year, 1, 29).getMonth() === 1;
+    }
+
+    function calculateAge(date) {
+      const d = new Date(date), now = new Date();
+      let years = now.getFullYear() - d.getFullYear();
+      d.setFullYear(d.getFullYear() + years);
+      if (d > now) {
+          years--;
+          d.setFullYear(d.getFullYear() - 1);
+      }
+      const days = (now.getTime() - d.getTime()) / (3600 * 24 * 1000);
+      return years + days / (isLeapYear(now.getFullYear()) ? 366 : 365);
+    }
+
+  
+    function defaultPatient() {
+      return {
+       'fname' : {'value': null},
+       'lname' : {'value': null},
+       'gender' : {'value': null},
+       'birthday' : {'value': null},
+       'age' : {'value': null}
+      };
+    }
+
+    function onError() {
+      console.log('Loading error', arguments);
+      ret.reject();
+    }
+
+    function onReady(smart)  {
+      const patient = smart.patient;
+      const pt = patient.read();
+      
+      $.when(pt).fail(onError);

```

Comment:
(This might be nitty gritty, but thought I would bring it up) One check we could consider is if the `smart` object has a `patient` property at all. Consider the scenario where the patient ID passed in the context is not given. In this case, it would get to `const pt = patient.read()` and throw an exception on the browser saying that it can't `read()` on `undefined`. So we can do a check like 
```javascript
if (smart.hasOwnProperty('patient')) { 
  \\ the rest of your code here
} else { 
  onError() 
}
```